bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java

```diff
@@ -381,6 +382,17 @@ public boolean start() throws IOException {
     @Override
     public List<SourceWithKeys> splitIntoBundles(
         long desiredBundleSizeBytes, PipelineOptions options) throws Exception {
+      // Update the desiredBundleSizeBytes in order to limit the
+      // number of splits to maximumNumberOfSplits.
+      long sizeEstimate = getEstimatedSizeBytes(options);
+      desiredBundleSizeBytes = Math.max(sizeEstimate / MAX_SPLIT_COUNT, desiredBundleSizeBytes);
```

Comment:
I see a few concerns, nothing major but atleast we need to document this (is a major concern if we should NEVER return more than MAX_SPLIT_COUNT to dataflow):
- This will still return at a minimum number of tables which could be > MAX_SPLIT_COUNT
- You are using the number of splits as (in split() function) as splitCount = (int) Math.ceil((double) (regionSize) / (double) (desiredBundleSizeBytes)); since it is the ceiling we will always get higher number of splits, say regionSize/desiredBundleSize is 0.75 then we will consistently return 2 splits per region and we will overflow MAX_SPLIT_SIZE.